controllers/cluster_mgmt_controller.go

```diff
@@ -0,0 +1,192 @@
+type ClusterMgmtController struct {
+	client.Client
+	logger      logr.Logger
+	ImageVector imagevector.ImageVector
+}
+
+// NewClusterMgmtController creates a new ClusterMgmtController object
+func NewClusterMgmtController(mgr manager.Manager) *ClusterMgmtController {
+	return &ClusterMgmtController{
+		Client: mgr.GetClient(),
+		logger: log.Log.WithName("cluster-mgmt-controller"),
+	}
+}
+
+// SetupWithManager sets up manager with a new controller and cmc as the reconcile.Reconciler
+func (cmc *ClusterMgmtController) SetupWithManager(mgr ctrl.Manager, workers int) error {
+
+	ctrl, err := controller.New(clusterMgmtControllerName, mgr, controller.Options{
+		Reconciler:              cmc,
+		MaxConcurrentReconciles: workers,
+	})
+	if err != nil {
+		return err
+	}
+
+	return ctrl.Watch(
+		&source.Kind{Type: &coordinationv1.Lease{}},
+		&handler.EnqueueRequestForOwner{OwnerType: &druidv1alpha1.Etcd{}, IsController: true},
+		// druidpredicates.LeaseHolderIdentityChange(),
+		druidpredicates.IsMemberLease(),
+	)
+}
+
+// +kubebuilder:rbac:groups=druid.gardener.cloud,resources=etcds,verbs=get;list;watch
+// +kubebuilder:rbac:groups=coordination.k8s.io,resources=leases,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile reconciles the multinode ETCD cluster.
+func (cmc *ClusterMgmtController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	cmc.logger.Info("Cluster management controller reconciliation started")
+	etcd := &druidv1alpha1.Etcd{}
+	if err := cmc.Get(ctx, req.NamespacedName, etcd); err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, return. Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{
+			RequeueAfter: 10 * time.Second,
+		}, err
+	}
+
+	logger := cmc.logger.WithValues("etcd", kutil.Key(etcd.Namespace, etcd.Name).String())
+
+	// run a loop every 5 minutes that will monitor the cluster health and take action if members in the etcd cluster are down
+	for {
+		if !etcd.DeletionTimestamp.IsZero() {
+			return ctrl.Result{Requeue: false}, nil
+		}
+
+		unknownThreshold := 300 * time.Second
+		notReadyThreshold := 60 * time.Second
+
+		checker := etcdmember.ReadyCheck(cmc.Client, logger, controllersconfig.EtcdCustodianController{
+			EtcdMember: controllersconfig.EtcdMemberConfig{
+				EtcdMemberNotReadyThreshold: notReadyThreshold,
+				EtcdMemberUnknownThreshold:  unknownThreshold,
+			},
+		})
+
+		results := checker.Check(context.Background(), *etcd)
+
+		totalReadyMembers := 0
+
+		for _, result := range results {
+			if result.Status() == "LeaseSucceeded" {
+				totalReadyMembers = totalReadyMembers + 1
+			}
+		}
+
+		quorum := int(etcd.Spec.Replicas)/2 + 1
+
+		if totalReadyMembers < quorum {
+			// scale down the statefulset to 0
+			sts := &appsv1.StatefulSet{}
```

Comment:
**Case 1**
when `readyCheck.Check` is invoked then then members which are currently in bootstrapping phase are not considered and therefore not added to the []Result. In this case the `totalReadyMembers < quorum` will be true and it will try and scale down sts when there are already members that are in bootstrapping phase. So yes there is a quorum loss but should we not wait for the bootstrapping to finish and the member to join the cluster instead of scaling down the entire cluster to 0?

**Case 2**
There was an error fetching leases. In this case in `readyCheck.Check` an empty []Result will be returned. In this case also is it prudent to scale down the entire sts to 0?

**Case 3**
In `readyCheck.Check` members whose containers are not yet ready are added with status `druidv1alpha1.EtcdMemberStatusNotReady`. In this case also will it be prudent to scale down the sts cluster to 0?

Similarly please consider all cases where an action to scale down sts would make sense.